src/prefect/server/database/orm_models.py

```diff
@@ -1399,7 +1416,14 @@ def __tablename__(cls) -> str:
         return "events"
 
     __table_args__: Any = (
-        sa.Index("ix_events__related_resource_ids", "related_resource_ids"),
+        sa.Index("ix_events__related_resource_ids", "related_resource_ids").ddl_if(
+            dialect="sqlite"
+        ),
+        sa.Index(
+            "ix_events__related_resource_ids_md5",
+            sa.text("md5(related_resource_ids::text)"),
+            postgresql_using="btree",
+        ).ddl_if(dialect="postgresql"),
```

Comment:
When could this be used?  Unless we're updating the queries to use `where related_resource_ids = md5(...)`, then I don't think it will be in play?  And even then, I think an exact match on the resources IDs would be a pretty rare query.  Maybe just remove the index for now?

Reply:
It gets used if you filter events by related resource ID. Postgres also suggests using a full-text index which might work better? 
```
2025-04-04 09:05:04.620 UTC [129] HINT:  Values larger than 1/3 of a buffer page cannot be indexed.
        Consider a function index of an MD5 hash of the value, or use full text indexing.
```

Reply:
@chrisguidry I updated the existing indexes to take advantage of GIN, which also seems to fix the original issue. Could you take a look at the new indexes?